src/main/java/com/yelp/nrtsearch/server/luceneserver/SimpleCopyJob.java

```diff
@@ -56,19 +55,8 @@ public SimpleCopyJob(
 
   @Override
   protected CopyOneFile newCopyOneFile(CopyOneFile prev) {
-    Iterator<RawFileChunk> rawFileChunkIterator;
-    try {
-      rawFileChunkIterator = primaryAddres.recvRawFile(prev.name, prev.getBytesCopied(), indexName);
-    } catch (Throwable t) {
-      try {
-        cancel("exc during start", t);
-      } catch (IOException e) {
-        throw new NodeCommunicationException("cancel IOException during newCopyOneFile", e);
-      }
-      throw new NodeCommunicationException("exc during start", t);
-    }
-
-    return new CopyOneFile(prev, rawFileChunkIterator);
+    // no state needs to be changed when transferring to a new job
+    return prev;
```

Comment:
Should we check if `prev` is closed, and if it is we create a new `CopyOneFile` ? Just wondering if Lucene can change the code to close `current` before trying to create a new `CopyOneFile`

Reply:
The main purpose of this method is to transfer the in progress copy state, in place of closing the file copy. If prev was closed, even the existing code would fail since the index file would also be closed.

Reply:
I think this change should work.

[This](https://github.com/apache/lucene-solr/blob/1d85cd783863f75cea133fb9c452302214165a4d/lucene/replicator/src/test/org/apache/lucene/replicator/nrt/SimpleCopyJob.java#L192) is the lucene test code I based this code off. 

Seems like the original test code simply made a new `CopyOneFile` object but re-used the `DataInput` i.e. made a shallow copy only.